src/audio/dai.c

```diff
@@ -32,20 +32,17 @@
 #define trace_dai(__e, ...)					\
 	trace_event(TRACE_CLASS_DAI, __e, ##__VA_ARGS__)
 #define trace_dai_with_ids(comp_ptr, __e, ...)			\
-	trace_event_comp(TRACE_CLASS_DAI, comp_ptr,		\
-			 __e, ##__VA_ARGS__)
+	trace_event_comp(DAI, comp_ptr, __e, ##__VA_ARGS__)
```

Comment:
Wait, so this "DAI" is not defined but due to how it is used it's actually concatenated to TRACE_CLASS_DAI later on? I find that slightly confusing/sneaky.

Reply:
But I do see a reasoning as well, to not have it automatically expand to the trace class number itself. I wonder if there is a way to refactor this in another way though.

Reply:
@paulstelian97 It would have worked properly, if it wasn't for `trace_event_comp` proxy macro, causing premature expansion, as you've noticed. I could've made it work with `META_DEFER` macros, iff trace classes were defined with _()_. But I figured that changing trace class definitions would cause much more hassle, than a little quirky concatenation under the hood.

Reply:
I see. Then I don't see anything that can be easily solved so I'll approve now.